helpers.go

```diff
@@ -153,10 +153,17 @@ func goType(pkg string, f *descriptor.FieldDescriptorProto) string {
 	case descriptor.FieldDescriptorProto_TYPE_MESSAGE:
 		if pkg != "" {
 			pkg = pkg + "."
+		} else {
+			pkg = getPackageTypeName(*f.TypeName) + "."
 		}
+
 		if *f.Label == descriptor.FieldDescriptorProto_LABEL_REPEATED {
 			return fmt.Sprintf("[]*%s%s", pkg, shortType(*f.TypeName))
 		}
+
+		if *f.Label == descriptor.FieldDescriptorProto_LABEL_REPEATED {
```

Comment:
If I'm not mistaken, above if statement is exactly the same as this one, and if the above if is true we never reach this one below due to the return statement.